### PR TITLE
Resolves the bugs associated with presenting multiple routes in succession within a single hit

### PR DESCRIPTION
**Pull request by maddalihanumateja**
Procedure for testing is similar to #972 . 

- Visit the url http://localhost:9000/?assignmentId=some_string&workerId=some_string&hitId=some_string&turkSubmitTo=localhost:5000/post . This will automatically assign you to a condition id. (or manually assign yourself a specific condition on the auditCondition endpoint http://localhost:9000/auditCondition?conditionId=some_condition&assignmentId=some_string&workerId=some_string&hitId=some_string&turkSubmitTo=localhost:5000/post)
- Check that the turker table inserts a new record with your workerId and conditionId
- Check that a record is inserted in the amt_assignment table with a route_id that is correctly associated with the assigned condition_id.
- Check that the distance to be audited in the mission is 1000ft for the first and second missions and 2000ft (displayed as 1/2 mile for some reason) for the third mission.
- Check that the mission progress bar updation is accurate. (For example, it shows 0% at the start, progresses quickly for a 1000ft mission vs a 2000ft mission, and is around 100% at completion).
- Finish the mission associated with the route and click on the "Continue" button. This should take you to the next route. (Check if it stays on the same route and goes into a loop where every step just triggers another mission complete event till you reach the end of the street edge.)
- Check that labels that are placed on the front end are stored in the label table.
- At the end of every mission check that the completed field for that route is changed to true in the amt_assignment table. Also check that a new record was inserted into the amt_assignment table with the route_id of the second mission and that this route_id associated with the condition assigned to workerId.
- Continue auditing routes till you see the "Submit HIT" button. This indicates that you have completed all routes that were available to you. Clicking this button shouldn't produce any visible change.

After some investigation it was found that there were bugs when a route ended on a particular street edge e1 and the next route started on a different edge e2. The bug arises because the current route was being set to the next route and the current edge was not being updated to the first edge of the next route. I've fixed this.

### CODE REVIEW

File: public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js [hunk 1]
```diff
@@ -149,26 +149,39 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
...
         var url = "/audit/amtAssignment ";
 
         // Fetch tasks for the route
-        taskContainer.fetchTasksOnARoute(nextMission.getProperty("routeId"), function () {
+        taskContainer.fetchTasksOnARoute(nextRouteId, function () {
             console.log("Tasks for the next route have been fetched");
+            // Replace current task with the first task from the next route
+            var newTaskId = Object.keys(taskContainer._taskStoreByRouteId[nextRouteId]).filter(function start(el){return taskContainer._taskStoreByRouteId[nextRouteId][el]['isStartEdge'];})[0];
+            var newTask = taskContainer._taskStoreByRouteId[nextRouteId][newTaskId].task;
+            var currentTaskId = taskContainer.getCurrentTask().getStreetEdgeId();
```
Comment (manaswis) [inline public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js]: Variable name needs updation

File: public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js [hunk 2]
```diff
@@ -149,26 +149,39 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
...
         svl.routeContainer.add(route);
         svl.routeContainer.setCurrentRoute(route);
         var url = "/audit/amtAssignment ";
 
         // Fetch tasks for the route
-        taskContainer.fetchTasksOnARoute(nextMission.getProperty("routeId"), function () {
+        taskContainer.fetchTasksOnARoute(nextRouteId, function () {
             console.log("Tasks for the next route have been fetched");
+            // Replace current task with the first task from the next route
+            var newTaskId = Object.keys(taskContainer._taskStoreByRouteId[nextRouteId]).filter(function start(el){return taskContainer._taskStoreByRouteId[nextRouteId][el]['isStartEdge'];})[0];
```
Comment (manaswis) [inline public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js]: Update this variable name as well.

**Comment by maddalihanumateja** [comment]
Updated the variable names.

File: public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js [hunk 3]
```diff
@@ -149,26 +149,39 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
         //Added code here to bring up the submit HIT button and post to the turkSubmit link
         // Or just trigger an event here such that the form submission (POST request to turkSubmit) happens on this event
         if (nextMission == null) {
+            _modalModel.updateModalMissionComplete(currentMission, currentNeighborhood);
+            _modalModel.showModalMissionComplete();
             _modalModel.showModalMissionCompleteHITSubmission();
             throw new Error("No missions available");
```
Comment (manaswis) [inline public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js]: use console.error here

Comment (maddalihanumateja) [inline public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js]: done

**Pull request merged**